tests/ToolTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Tightenco\NovaGoogleAnalytics\Tests;
+
+use Tightenco\NovaGoogleAnalytics\Http\Controllers\GoogleAnalyticsController;
+use Spatie\Analytics\Period;
+
+class ToolTest extends TestCase
+{
+    /** @test */
+    public function it_can_return_a_response()
+    {
+        $this
+            ->get('nova-vendor/tightenco/nova-google-analytics/pages')
+            ->assertSuccessful()
+            ->assertJsonStructure([
+                'hasMore',
+                'pages',
+                'totalPages'
+            ]);
+    }
+
+    /** @test */
+    public function it_will_accept_limit_param()
+    {
+        $this
+            ->get('nova-vendor/tightenco/nova-google-analytics/pages?limit=5')
+            ->assertJsonCount(5, 'pages');
+    }
+
+    /** @test */
+    public function it_will_not_accept_non_numeric_limit()
+    {
+        $this->get('nova-vendor/tightenco/nova-google-analytics/pages?limit=a')
+            ->assertStatus(500);
+    }
+
+    /** @test */
+    public function it_will_accept_page_param()
+    {
+        $page1 = $this->get('nova-vendor/tightenco/nova-google-analytics/pages?page=1');
+        $page2 = $this->get('nova-vendor/tightenco/nova-google-analytics/pages?page=2');
+
+        $this->assertNotEquals($page1, $page2);
+    }
+
+    /** @test */
+    public function it_will_not_accept_non_numeric_page()
+    {
+        $this->get('nova-vendor/tightenco/nova-google-analytics/pages?page=a')
+            ->assertStatus(500);
+    }
+
+    /** @test */
+    public function it_will_accept_duration_param()
+    {
+        $this->get('nova-vendor/tightenco/nova-google-analytics/pages?duration=month')
+            ->assertSuccessful();
+    }
+
+    /** @test */
+    public function invalid_duration_will_default_to_week()
+    {
+        $controller = new GoogleAnalyticsController();
+        $reflection = new \ReflectionClass($controller);
+        $method = $reflection->getMethod('periodForDuration');
+        $method->setAccessible(true);
+        $duration = $method->invokeArgs($controller, ['asdf']);
+        $this->assertEquals(Period::days(7), $duration);
+    }
```

Comment:
Nice work figuring this out! Normally, if something needs testing, we avoid doing this sort of a hacky test and instead moving the logic out of the controller and into a configuration or class that's public and testable. In general, I like to never instantiate a controller in a test.

In this case, I think we may just want to create a custom class that has the responsible of taking the parameters from the controller and doing basically all the work the controller is doing right now, so that A) the controller is cleaner and B) we can just comfortably make these methods public to test them.